src/main/java/core/basesyntax/MatrixOperation.java

```diff
@@ -6,18 +6,38 @@
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int[] diagonal = new int[matrix.length];
+        for (int i = 0; i < matrix.length; i++) {
+            diagonal[i] = matrix[i][i];
+        }
+        return diagonal;
     }
 
     public int[] getCounterDiagonal(int[][] matrix) {
-        return null;
+        int[] counterDiagonal = new int[matrix.length];
+        for (int i = 0; i < matrix.length; i++) {
+            counterDiagonal[i] = matrix[i][matrix.length - 1 - i];
+        }
+        return counterDiagonal;
     }
 
     public int getMin(int[][] matrix) {
-        return 0;
+        int min = matrix[0][0];
+        for (int[] innerArray : matrix) {
+            for (int data : innerArray) {
+                min = Math.min(min, data);
```

Comment:
read checklist attentively https://mate-academy.github.io/jv-program-fulltime/01_core/checklist/hw_02.html
- Find min and max element without Math library